website/docs/concepts/why_immutability.mdx

```diff
@@ -0,0 +1,125 @@
+---
+title: Why Immutability
```

Comment:
This article explains what is immutability and how to use it, but it doesn't seem to explain why use it and how it compares with mutability.

Reply:
Good point, I'll work on this later this week.